Gems/Terrain/Code/Source/TerrainRenderer/TerrainDetailMaterialManager.cpp

```diff
@@ -190,39 +186,13 @@ namespace Terrain
         return IndicesValid && m_detailMaterialDataBuffer.IsValid();
     }
     
-    void TerrainDetailMaterialManager::RemoveAllImages()
-    {   
-        for (const DetailMaterialData& materialData: m_detailMaterials.GetDataVector())
-        {
-            DetailMaterialShaderData& shaderData = m_detailMaterialShaderData.GetElement(materialData.m_detailMaterialBufferIndex);
-
-            auto checkRemoveImage = [&](uint16_t index)
-            {
-                if (index != 0xFFFF)
-                {
-                    m_bindlessImageHandler->RemoveBindlessImage(index);
-                }
-            };
-            
-            checkRemoveImage(shaderData.m_colorImageIndex);
-            checkRemoveImage(shaderData.m_normalImageIndex);
-            checkRemoveImage(shaderData.m_roughnessImageIndex);
-            checkRemoveImage(shaderData.m_metalnessImageIndex);
-            checkRemoveImage(shaderData.m_specularF0ImageIndex);
-            checkRemoveImage(shaderData.m_occlusionImageIndex);
-            checkRemoveImage(shaderData.m_heightImageIndex);
-        }
```

Comment:
Do this still need to do a reference decrement on the images? Or am I misunderstanding this function?

Reply:
The old function removed the image from the bindless array that terrain controlled, but the new system will just drop the image automatically whenever the image view dies, so terrain doesn't need to do anything.